phi/tools/firecrawl.py

```diff
@@ -13,29 +16,39 @@ class FirecrawlTools(Toolkit):
     def __init__(
         self,
         api_key: Optional[str] = None,
+        api_url: Optional[str] = None,
         formats: Optional[List[str]] = None,
-        limit: int = 10,
+        crawl_limit: int = 10,
         scrape: bool = True,
         crawl: bool = False,
+        async_crawl: bool = False,
+        map: bool = False,
+        map_limit: int = 10,
     ):
         super().__init__(name="firecrawl_tools")
 
-        self.api_key: Optional[str] = api_key
+        self.api_key: Optional[str] = api_key or os.getenv("FIRECRAWL_API_KEY")
+        self.api_url: Optional[str] = api_url or os.getenv("FIRECRAWL_API_URL", "https://api.firecrawl.dev")
         self.formats: Optional[List[str]] = formats
-        self.limit: int = limit
+        self.crawl_limit: int = crawl_limit
+        self.map_limit: int = map_limit
```

Comment:
we want backward compatibility. So keep the limit for now and write a comment to deprecate later. Your logic should also use limit param

Reply:
makes sense. will make the required changes.